application/lib/view_model/sign_up_view_model.dart

```diff
@@ -0,0 +1,98 @@
+import 'package:application/model/user_model.dart';
+import 'package:application/utils/roots.dart';
+import 'package:application/utils/translation_en.dart';
+import 'package:application/view/screens/base_screen.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+class SignUpViewModel extends ChangeNotifier {
+  final FirebaseAuth _firebase = FirebaseAuth.instance;
+
+  bool _isLoading = false;
+
+  bool get isLoading => _isLoading;
+
+  set isLoading(bool value) {
+    _isLoading = value;
+    notifyListeners();
+  }
+
+  Future<bool> submitSignUp(String enteredFirstName, String enteredEmail,
```

Comment:
A Firebase authentikáció és ez a függvény nem a viewmodel felelősségi köre. A viewmodel feladata, hogy az üzleti logikát kezelje, tehát tudja, hogy mikor melyik funkciót kell végrehajtani és a funkciókhoz melyik service-t kell meghívni. Amit küldtem minta, ott is látszódik, hogy van még egy plusz service layer. Legyen egy AuthenticationService.dart és ez a service legyen felelős az authentikációért. Ez a submitSignUp függvény pedig hívja az AuthenticationService megfelelő függvényét. Miért csináljuk ezt így? Azért hogy bármelyik réteg könnyen cserélhető legyen. Tehát ha egyszer úgy dönt az ember hogy firebase helyett másik adatbázis kezelőt használ, akkor ne kelljen az összes viewmodelben hosszasan újra kódolni, hanem csak a service fileokat kicseréljük új service fileokra, amikben implementálva vannak a szükséges függvények, és a viewmodel az új servicet filet hívja be.